test/functional/p2p_segwit.py

```diff
@@ -550,37 +535,9 @@ def test_v0_outputs_arent_spendable(self):
             # TODO: support multiple acceptable reject reasons.
             test_witness_block(self.nodes[0], self.test_node, block, accepted=False, with_witness=False)
 
-        self.connect_nodes(0, 2)
-
         self.utxo.pop(0)
         self.utxo.append(UTXO(txid, 2, value))
 
-    @subtest  # type: ignore
-    def test_getblocktemplate_before_lockin(self):
-        txid = int(self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 1), 16)
-
-        for node in [self.nodes[0], self.nodes[2]]:
-            gbt_results = node.getblocktemplate({"rules": ["segwit"]})
-            if node == self.nodes[2]:
-                # If this is a non-segwit node, we should not get a witness
-                # commitment.
-                assert 'default_witness_commitment' not in gbt_results
-            else:
-                # For segwit-aware nodes, check the witness
-                # commitment is correct.
-                assert 'default_witness_commitment' in gbt_results
-                witness_commitment = gbt_results['default_witness_commitment']
-
-                # Check that default_witness_commitment is present.
-                witness_root = CBlock.get_merkle_root([ser_uint256(0),
-                                                       ser_uint256(txid)])
-                script = get_witness_script(witness_root, 0)
-                assert_equal(witness_commitment, script.hex())
```

Comment:
why is the test for default_witness_commitment removed?

Reply:
Segwit is a buried deployment. There are no miners calling `getblocktemplate` before the segwit activation is locked in.

Reply:
The commitment is produced regardless of whether segwit locked in or not.

Reply:
Oh, I see now that `default_witness_commitment` is not tested anywhere else. Perhaps it should be tested in mining_basic.py?

Reply:
Thank you for the catch, @MarcoFalke ! Test is now moved into mining_basic.py